### PR TITLE
storage: Support "noauto" crypto backing devices better

### LINKED ISSUE #14467: Unnecessary notification - Warning alert:Inconsistent filesystem mount
Cockpit version: 225
OS: Debian 10 (Buster)
Page: Storage

I'm running a headless server with encrypted partitions. The root file system is not encrypted.
I don't like to have the passwords for the encrypted file systems stored on the server itself, so after a reboot, I manually unlock the encrypted file systems and mount them (both via Cockpit).
In order to prevent the server from booting into emergency mode, I have to give the custom mount option 'noauto' (I have to type this every time I mount the file system, which is annoying as well).
This approach results in the "Warning alert:Inconsistent filesystem mount".

I think this is a valid use case of the system, and therefore the warning should be considered unnecessary.

If I am incorrectly using the system and there is another way to organize the encrypted partitions without storing the password on the server itself, I would be glad to rearrange my system accordingly.

### PR DESCRIPTION
**Pull request by mvollmer**
A filesystem that has a crypto backing device with the "noauto" option
will now always have the "noauto" mount option, even when mounted.
    
If such a filesystem doesn't have the "noauto" option, Cockpit will
warn about it and offer to add it.

Fixes #14467

#### Storage: Better support for "noauto" LUKS devices

Cockpit now behaves more sensible with encrypted filesystems that have "noauto" in their LUKS options. Such a filesystem will now also always have "noauto" in their mounting options even when mounted.

### CODE REVIEW

**Comment by mvollmer** [comment]
Test is missing.

**Comment by mvollmer** [comment]
The code is getting quite horrible, but #13192 will redo all this and hopefully clean this up.

**Comment by mvollmer** [comment]
> Test is missing.

Added.

**Comment by mvollmer** [comment]
Hmm, we don't actually need to show "noauto" in the "custom options" field.  We can just keep managing it implicitly, and just always enable it when the luks container is noauto.

**Comment by mvollmer** [comment]
> Hmm, we don't actually need to show "noauto" in the "custom options" field. We can just keep managing it implicitly, and just always enable it when the luks container is noauto.

Done, much nicer IMO.

**Comment by mvollmer** [comment]
I broke all the tests.

File: pkg/storaged/fsys-tab.jsx [hunk 1]
```diff
@@ -137,14 +154,12 @@ export function mounting_dialog(client, block, mode) {
         var new_config = null;
         var all_new_opts;
 
-        if (old_config) {
```
Comment (marusak) [inline pkg/storaged/fsys-tab.jsx:140]: This seems unrelated and possibly wrong? (not sure though why we needed the check in the first place)

File: test/verify/check-storage-luks [hunk 2]
```diff
@@ -377,5 +377,59 @@ class TestStorageLuks(StorageCase):
...
+                     "passphrase2": "vainu-reku-toma-rolle-kaja",
+                     "mount_options.auto": True,
+                     "mount_point": "/run/foo"})
+        self.content_row_wait_in_col(1, 1, "Encrypted data")
+        self.content_tab_wait_in_info(1, 1, "Options", "noauto")
+        self.content_row_wait_in_col(2, 1, "ext4 File System")
+
+        # The filesystem should be mounted but have the "noauto" option
+        self.wait_mounted(2,1)
+        assert "noauto" in m.execute("grep /run/foo /etc/fstab || true")
```
Comment (marusak) [inline test/verify/check-storage-luks:407]: interesting, had no idea that this kind of assert is possible. But we use it a few times in verify tests indeed. Not sure if there is better/worse backtrace, but one can use `self.assertIn("noauto", m.execute("grep /run/foo /etc/fstab || true"))

File: pkg/storaged/fsys-tab.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/fsys-tab.jsx:140]: This is indeed non-obvious, but I'd say the code was buggy earlier.  With the check, we could not set any options when mounting a filesystem that wasn't listed in /etc/fstab.

When adding an entry to /etcfstab with this function (as opposed to changing an existing one), it would always end up having "default" options. Now we want to add "noauto" in certain cases, and that's how I noticed.

I guess I should add a test for this...

File: test/verify/check-storage-luks (hunk 2 above)
Comment (mvollmer) [reply test/verify/check-storage-luks:407]: Yes, I'll change that to `self.assertIn`.

Comment (mvollmer) [reply test/verify/check-storage-luks]: Done.

File: pkg/storaged/fsys-tab.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/fsys-tab.jsx:140]: Done.

**Review by marusak — APPROVED** [review]
Thanks!
